### DISCUSSION TITLE
Generic Java library for Keycloak

### DISCUSSION THREAD

**Discussion by vignesh-manel** [keycloak/keycloak#21390] [Q&A]
I am working on a java library where keycloak will be used for authentication/authorization. I want this to be generic without depending on underlying framework/server like spring/jboss etc. Which java library can be used here? Because as per docs all libraries are dependent of specific implementation - https://www.keycloak.org/docs/latest/securing_apps/index.html#client-adapters
And in this I see a generic java library will be made available in keycloak 22 - https://www.keycloak.org/2023/03/adapter-deprecation-update
So currently is there no vanilla java library which can used to integrate with keycloak without a specifc underlying implementation?

**Comment by vignesh-manel** [comment]
Hi, Any inputs on this?

**Reply by danielFesenmeyer** [reply]
I would suggest to have a look at this list: https://oauth.net/code/java/

To be honest, the only one of those I have really used was scribejava, and I wasn't satisfied with it - the main issue were several old issues which were never resolved (e.g. for keycloak configuration).
Nimbus is used by Spring internally and seems to be quite mature, maybe that could make sense for you.

**Reply by embesozzi** [reply]
I tried [nimbus](https://connect2id.com/products/nimbus-oauth-openid-connect-sdk) in the past with a custom OIDC Filter and it was quite good. But then I moved to Spring Security.
